### DISCUSSION TITLE
Custom Livewire Attributes

### DISCUSSION THREAD

**Discussion by pizavo** [livewire/livewire#8698] [3. Feature ideas]
Hello there,
I'd like to ask, would it be possible to allow for custom livewire attributes? Right know, only `#[Title]` and `#[Layout]` are hardcoded into the system, but I think many people could leverage the ability to add custom attributes, for me it would be the `#[Description]` attribute, which has very similar use-case to the `#[Title]` attribute and I don't see it very elegant to always add it to `view.layoutData` function, when the whole `render` function in the component is optional.
Thanks for reply.

**Comment by Fludem** [comment]
What would your use for the description attribute be exactly?\

**Reply by pizavo** [reply]
The same as `#[Title]`, basically. `#[Title]` enables to pass `$title` into the layout, but `<title>` of the page is not the only thing that could possibly change in the layout with change of the Livewire page component. That's where `#[Description]` comes into play, as it would be possible to pass `$description` into the `<meta>` tags in the `<head>`. Right now, I solve this with a scoped service, however, I don't find this solution elegant. Having the ability to register custom _`LivewireAttribute`_ would be much helpful, as it might be used for different use-cases as well.

**Comment by danie-ramdhani** [comment]
Yes you can create custom livewire attributes using artisan command

```sh
php artisan livewire:attribute SomeAttribute
```

**Reply by pizavo** [reply]
Yeah, but there is no guide on how to use it. I already tried that and it didn't work. Annotating the class or the `render` function with said `#[Description]` attribute didn't work, the `$description` variable in the layout threw an exception, as it was not defined, which makes sense, as the `interceptTheRenderOfTheComponentAndRetreiveTheLayoutConfiguration` function does not register it (or at least that's the reason I came up with, as these frameworks are really complex and who knows how undocumented things really work, speaking of the command).

**Reply by lrljoe** [reply]
What does your Description attribute contain?
It should have a View macro etc

e.g, the title uses this, you'll obviously need to implement some hooks/methods

```php
        View::macro('title', function ($title) {
            if (! isset($this->layoutConfig)) $this->layoutConfig = new PageComponentConfig;

            $this->layoutConfig->mergeParams(['title' => $title]);

            return $this;
        });
```

**Reply by pizavo** [reply]
@lrljoe Oh, I didn't know that, I simply tried to copy the `BaseTitle` which doesn't have anything like that. Is there any guide, docs or anything on how to implement it properly?

**Reply by lrljoe** [reply]
There isn't a guide at the moment as far as I am aware
